smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelValidator.java

```diff
@@ -214,6 +245,39 @@ public Validator createValidator() {
         };
     }
 
+    /**
+     * Applies each of the given decorators to each of the validation events and returns the resulting list of
+     * validation events.
+     *
+     * @param decorators The decorators to apply
+     * @param events The events to decorate
+     * @return The decorated events
+     */
+    public static List<ValidationEvent> decorateEvents(
+        List<ValidationEventDecorator> decorators,
+        List<ValidationEvent> events
+    ) {
+        if (!decorators.isEmpty()) {
+            try {
+                List<ValidationEvent> decorated = new ArrayList<>(events.size());
```

Comment:
Can we assume the array is mutable? If the list has a size > 0, I think we can, right? It's not like this method is getting called by something we don't control since it's package private.

Reply:
Yeah, I think we can assume that, let me change the code to update the slot instead of copying the values.

Reply:
Changed to use the original list overwriting the value in place.